File: dotnet/src/MyDotey.SCF/DefaultConfigurationSourceChangeEvent.cs
using System;

namespace MyDotey.SCF
{
    /**
     * @author koqizhao
     *
     * Jul 19, 2018
     */
    public class DefaultConfigurationSourceChangeEvent : IConfigurationSourceChangeEvent
    {
        public virtual IConfigurationSource Source { get; protected set; }
        public virtual long ChangeTime { get; protected set; }

        public DefaultConfigurationSourceChangeEvent(IConfigurationSource source)
            : this(source, DateTime.Now.Ticks / TimeSpan.TicksPerMillisecond)
        {

        }

        public DefaultConfigurationSourceChangeEvent(IConfigurationSource source, long changeTime)
        {
            if (source == null)
                throw new ArgumentNullException("source is null");

            Source = source;
            ChangeTime = changeTime;
        }

        public override String ToString()
        {
            return string.Format("{0} {{ source: {1}, changeTime: {2} }}", GetType().Name, Source, ChangeTime);
        }
    }
}

File: java/scf-core/src/main/java/org/mydotey/scf/DefaultConfigurationSourceChangeEvent.java
package org.mydotey.scf;

import java.util.Objects;

/**
 * @author koqizhao
 *
 * Jul 19, 2018
 */
public class DefaultConfigurationSourceChangeEvent implements ConfigurationSourceChangeEvent {

    private ConfigurationSource _source;
    private long _changeTime;

    public DefaultConfigurationSourceChangeEvent(ConfigurationSource source) {
        this(source, System.currentTimeMillis());
    }

    public DefaultConfigurationSourceChangeEvent(ConfigurationSource source, long changeTime) {
        Objects.requireNonNull(source, "source is null");
        _source = source;
        _changeTime = changeTime;
    }

    @Override
    public ConfigurationSource getSource() {
        return _source;
    }

    @Override
    public long getChangeTime() {
        return _changeTime;
    }

    @Override
    public String toString() {
        return String.format("%s { source: %s, changeTime: %s }", getClass().getSimpleName(), _source, _changeTime);
    }

}
